pysal/contrib/viz/plot.py

```diff
@@ -0,0 +1,47 @@
+"""
+Moran Plot using PySAL and Matplotlib
+
+"""
+
+__author__ = "Marynia Kolak <marynia.kolak@gmail.com>"
+
+import pandas as pd
+import numpy as np
+import pysal as ps
+import matplotlib.pyplot as plt
+
+
+def Moran(var, w, **kwargs):
```

Comment:
At this point, I'd be inclined to repeat computation in the name of keeping things modular and API-clean (e.g. taking in PySAL objects). If we find it a bottleneck, we can always pass out `zx` and `zy` from `Moran`, it wouldn't break the existing API. If the user's past everything needed for the plot to `Moran`, that should be all they need to pass, imho. Also, `zx` and `zy` are not likely to be the bottlenecks on this (it's the permutations rather), which we wouldn't need.